Add procedures on how to import ansible playbooks (#1746)
theforeman/foreman-documentation: guides/common/modules/proc_importing-all-available-ansible-playbooks.adoc

[id="importing-all-available-ansible-playbooks_{context}"]
= Importing All Available Ansible Playbooks

> You can import all the available Ansible playbooks on your host.

Author: ```suggestion
You can import all the available Ansible playbooks from collections installed on the Capsule.
```

> You can import all the available Ansible playbooks from collections installed on the Capsule.

Author: ```suggestion
You can import all the available Ansible playbooks from collections installed on the {SmartProxy.
```

> You can import all the available Ansible playbooks from collections installed on the {SmartProxy}.

Reviewer: ```suggestion
You can import all the available Ansible playbooks to {Project} from collections installed on {SmartProxy}.
You can then use the playbooks to invoke jobs using the _Ansible - Run playbook_ job template.
```

.Procedure

> . On the target host, import the Ansible playbooks:

Reviewer: ```suggestion
. Import the Ansible playbooks using the following API request:
```
The API call does not have to be made from a host and it does not relate to a particular host. (API calls can be technicaly made from a local machine or anywhere that can access the SatServer over HTTPS.)

[options="nowrap", subs="+quotes,verbatim,attributes"]
----

> curl -X PUT 'Content-Type: application/json' https://_{smartproxy-example-com}_:{smartproxy_port}/ansible/api/v2/ansible_playbooks/sync?proxy_id=_proxy_id_number_ | json_pp

Reviewer 2: I'm not entirely sure what smartproxy-example-com and smartproxy_port resolves to, but it's the Foreman's API and port right? Similarly in the other guide.

Reviewer 3: ```console
$ rg 'smartproxy-example-com|smartproxy_port' guides/common/attributes*
guides/common/attributes-satellite.adoc
87::smartproxy_port: 9090
88::smartproxy-example-com: capsule.example.com

guides/common/attributes-orcharhino.adoc
45::smartproxy-example-com: orcharhino-proxy.example.com

guides/common/attributes-katello.adoc
6::smartproxy_port: 9090

guides/common/attributes-base.adoc
98::smartproxy_port: 8443
99::smartproxy-example-com: smartproxy.example.com
```

Reviewer 3: It's very uncommon to connect to a Smart Proxy, especially when you also pass a proxy ID. Didn't you mean `{foreman-example-com}` (and leave out the port)?

Reviewer: Yes, the API call should be pointed at a SatServer URL.

[id="importing-all-available-ansible-playbooks_{context}"]
= Importing All Available Ansible Playbooks

You can import all the available Ansible playbooks to {Project} from collections installed on {SmartProxy}.

.Prerequisite
* Ansible plug-in in {Project} is enabled

.Procedure
. Import the Ansible playbooks using the following API request:
+
[options="nowrap", subs="+quotes,verbatim,attributes"]
----

> curl -X PUT 'Content-Type: application/json' https://_{foreman-example-com}_/ansible/api/v2/ansible_playbooks/sync?proxy_id=_My_{smart-proxy-context}_ID_ | json_pp

Author: ```suggestion
curl -X PUT 'Content-Type: application/json' https://_{foreman-example-com}_/ansible/api/v2/ansible_playbooks/sync?proxy_id=_My_{smart-proxy-context}_ID_
```

[id="importing-all-available-ansible-playbooks_{context}"]
= Importing All Available Ansible Playbooks

You can import all the available Ansible playbooks to {Project} from collections installed on {SmartProxy}.

.Prerequisite
* Ansible plug-in in {Project} is enabled

.Procedure

> . Import the Ansible playbooks using the following API request:

Reviewer 3: I was wondering: isn't this available in [hammer-cli-foreman-ansible](https://github.com/theforeman/hammer-cli-foreman-ansible). Looking at the source I only see one hit on `sync` (https://github.com/theforeman/hammer-cli-foreman-ansible/search?q=sync) and that's in https://github.com/theforeman/hammer-cli-foreman-ansible/blob/a2a884f1799e393f8a4ea340bc60c48dd8e2acc1/lib/hammer_cli_foreman_ansible/ansible_roles.rb#L28-L35

Given the filename and class name I get the impression it's only for roles, not playbooks. If we don't have this functionality, I think that deserves an issue to add it. Providing a CLI command is a better experience for most users than calling the API.

Reviewer: AFAIK, it's only available via API, not other UIs yet.

Reviewer 3: Bonus points, but in code I like to add comments for that. Something like
```suggestion
// Not available via Hammer/UI: <link-to-issue>
. Import the Ansible playbooks using the following API request:
```

Then in the future if you're reading it you know why it's done via the API and have an easy way to check if it's been resolved yet.

Author: > Then in the future if you're reading it you know why it's done via the API and have an easy way to check if it's been resolved yet.

Yeah, good idea

+
[options="nowrap", subs="+quotes,verbatim,attributes"]
----

> curl -X PUT 'Content-Type: application/json' https://_{foreman-example-com}_/ansible/api/v2/ansible_playbooks/sync?proxy_id=_My_smart-proxy-context_ID_ | json_pp

Reviewer: ```suggestion
curl -X PUT 'Content-Type: application/json' https://_{foreman-example-com}_/ansible/api/v2/ansible_playbooks/sync?proxy_id=_My_{smart-proxy-context}_ID_ | json_pp
```

> curl -X PUT 'Content-Type: application/json' https://_{foreman-example-com}_/ansible/api/v2/ansible_playbooks/sync?proxy_id=_My_{smart-proxy-context}_ID_ | json_pp

Reviewer 3: Where does `json_pp` come from? We don't have that anywhere in our current docs.

Author: > Where does `json_pp` come from? We don't have that anywhere in our current docs.

from the demo https://www.youtube.com/watch?v=ykQFM7aFczY&t=1788s

Reviewer 3: I think it's a developer short hand for JSON Pretty Print. Most people won't have that (on a related note, you can get the same by piping it to [jq](https://stedolan.github.io/jq/)). I'd leave it out
